ebt/Dialog/BtSender.vala

```diff
@@ -301,30 +310,31 @@ public class BtSender : Granite.Dialog {
         if (transfer_rate == 0) {
             return;
         }
-        rate_label.label = _("<b>Rate:</b> %s").printf (GLib.format_size (transfer_rate));
+        rate_label.label = _("<b>Transfer rate:</b> %s").printf (GLib.format_size (transfer_rate));
         uint64 remaining_time = (total_size - transferred) / transfer_rate;
-        progress_label.label = _("Sending… (%i/%i) %s of %s remaining %s").printf (current_file, total_file, GLib.format_size (transferred), GLib.format_size (total_size), format_time((int)remaining_time));
+        progress_label.label = _("Sending… (%i/%i) %s of Size: %s, remaining time %s").printf (current_file, total_file, GLib.format_size (transferred), GLib.format_size (total_size), format_time ((int)remaining_time));
     }
 
     private string format_time (int seconds) {
-	    int minutes;
-	    if (seconds < 0) {
-		    seconds = 0;
-		}
-	    if (seconds < 60) {
-		    return ngettext("%'d second", "%'d seconds", seconds).printf (seconds);
-		}
-	    if (seconds < 60 * 60) {
-		    minutes = (seconds + 30) / 60;
-		    return ngettext("%'d minute", "%'d minutes", minutes).printf (minutes);
-	    }
-	    int hours = seconds / (60 * 60);
-	    if (seconds < 60 * 60 * 4) {
-		    minutes = (seconds - hours * 60 * 60 + 30) / 60;
-		    string h = ngettext("%'u hour", "%'u hours", hours).printf (hours);
-		    string m = ngettext("%'u minute", "%'u minutes", minutes).printf (minutes);
-		    return h.concat(", ", m);
-	    }
-	    return ngettext("approximately %'d hour", "approximately %'d hours", hours).printf (hours);
+        int minutes;
+        if (seconds < 0) {
+            seconds = 0;
+        }
+        if (seconds < 60) {
+            return ngettext ("%'d second", "%'d seconds", seconds).printf (seconds);
+
+        }
+        if (seconds < 60 * 60) {
+            minutes = (seconds + 30) / 60;
+            return ngettext ("%'d minute", "%'d minutes", minutes).printf (minutes);
+        }
+        int hours = seconds / (60 * 60);
+        if (seconds < 60 * 60 * 4) {
+            minutes = (seconds - hours * 60 * 60 + 30) / 60;
+            string h = ngettext ("%'u hour", "%'u hours", hours).printf (hours);
+            string m = ngettext ("%'u minute", "%'u minutes", minutes).printf (minutes);
+            return h.concat (", ", m);
+        }
+        return ngettext ("approximately %'d hour", "approximately %'d hours", hours).printf (hours);
```

Comment:
This looks like repeated code - can you DRY it?  Maybe have a `Utils.vala` file?

Reply:
"approximately" used to display the initial progress that the file was received with an estimate later when the flat transfer speed was stable then will display the actual time

Reply:
I meant that the whole function is the same (or very similar) to the `format_time ()` in `BtReceiver.vala`